src/ui/chat-window.js

```diff
@@ -18,11 +18,86 @@ export class ChatScene extends Phaser.Scene {
 
     col1 = 0;
     visible = false;
+    isPromptOpen = false;
 
     constructor() {
         super({ key: CONSTANTS.SCENES.CHAT });
     }
 
+    // TODO: if we need more prompts we should reuse a lot of this
+    // Right now it is specifically for selling X amount of an item
+    prompt(promptText, item) {
+        const originallyVisible = this.visible;
+        this.show(false);
+        this.isPromptOpen = true;
+
+        let promptWindow = this.add
+            .image(0, 338, "prompt-window")
+            .setOrigin(0, 0)
+            .setDepth(8);
+
+        // TODO: Current x value is only for the text "Enter Amount:"
+        let prompt = this.add.text(225, 380, promptText, FONTS.PROMPT).setDepth(8);
+        let promptInput = this.add
+            .text(promptWindow.width / 2, 410, "*", FONTS.PROMPT_INPUT)
+            .setDepth(9);
+
+        this.input.keyboard.on("keydown", function (event) {
+            // enter
+            if (event.keyCode == 13) {
+                let inputValue = parseInt(
+                    promptInput.text.substr(1, promptInput.text.length)
+                );
+
+                if (isNaN(inputValue)) {
+                    inputValue = 0;
+                }
+
+                promptWindow.destroy();
+                prompt.destroy();
+                promptInput.destroy();
+                this.removeAllListeners();
+
+                item.sellX(inputValue);
+
+                this.scene.isPromptOpen = false;
+                this.scene.show(originallyVisible);
```

Comment:
If something is sold, it should write to chat saying so

Reply:
Updated! https://github.com/EthanVieira/rs-clicker/pull/56/commits/93c33cfc10a7f52b57dad5438451f5d67196d089